packages/paste-core/components/disclosure/__tests__/disclosure.test.tsx

```diff
@@ -164,6 +195,87 @@ describe('Disclosure', () => {
     });
   });
 
+  describe('Customization', () => {
+    it('should set an element data attribute for Disclosure components', () => {
+      render(
+        <Theme.Provider theme="default">
+          <MockDefaultElementDisclosure />
+        </Theme.Provider>
+      );
+
+      const renderedDisclosureHeading = screen.getByTestId('disclosure-heading');
+      const renderedDisclosure = renderedDisclosureHeading.parentElement.parentElement;
+      const renderedDisclosureContent = screen.getByTestId('disclosure-content');
+
+      expect(renderedDisclosure.getAttribute('data-paste-element')).toEqual('DISCLOSURE');
+      expect(renderedDisclosureHeading.getAttribute('data-paste-element')).toEqual('DISCLOSURE_HEADING');
+      expect(renderedDisclosureContent.getAttribute('data-paste-element')).toEqual('DISCLOSURE_CONTENT');
+    });
+
+    it('should set a custom element data attribute for custom named Disclosure components', () => {
+      render(
+        <Theme.Provider theme="default">
+          <MockCustomElementDisclosure />
+        </Theme.Provider>
+      );
+
+      const renderedDisclosureHeading = screen.getByTestId('disclosure-heading');
+      const renderedDisclosure = renderedDisclosureHeading.parentElement.parentElement;
```

Comment:
It may be better to give this a testid

Reply:
I tried to, put we don't spread BoxProps on the Disclosure prop, so it doesn't show up. I should've asked if that's on purpose or not actually. I assumed that it was for a specific reason, and this is my workaround for grabbing Disclosure. If not, I could just do safelySpreadBoxProps onto Disclosure and then use testId.

Reply:
Talked with Simon about it and implemented the the latter option (use safelySpreadBoxProps on Disclosure and getByTestId in tests). Thanks for pushing on this, @TheSisb!